### PR TITLE
Fixing any_unfulfilled_milestones in mobile course_enrollments API

### PR DESCRIPTION
**Pull request by ahmedaljazzar**
### Description

In [User Course Enrollments List](https://courses.edx.org/api/mobile/v0.5/users/<username>/course_enrollments), if we enabled **Entrance Exam** from _Schedule & Details_ section and then re-desabling it, the`courseware_access` field in the API will keep preventing the student from showing the course on the mobile apps. 

```JSON
            "courseware_access": {
                "has_access": false,
                "error_code": "unfulfilled_milestones",
                "developer_message": "User has unfulfilled milestones",
                "user_message": "User has unfulfilled milestones."
            },
```

This problem is happening because the `get_course_milestones_fulfillment_paths` method in the [milestones API](https://github.com/edx/edx-milestones/blob/master/milestones/api.py#L176) is returning an empty dictionary `{}` if the course doesn't have any milestones, after enabling *Entrance Exam*  it returns a complex dict-of-dicts like this:

```JSON
{
  "course-v1:org+course+run.entrance_exams.Completed Course Entrance Exam": {
     "content": [
          "block-v1:org+course+run+type@chapter+block@cd5f1b8bef874bcdad80b2857f5aeffd"
     ]
  }
}
```

While redoubling the entrance exam causes the method to return the previously-enabled prerequisites with an empty dictionary like this:

```JSON
{"course-v1:org+course+run.entrance_exams.Completed Course Entrance Exam": {}}
```

The method `any_unfulfilled_milestones` is not addressing this scenario, so what I did is checking that every value in the dict returned from `get_course_milestones_fulfillment_paths`, if any value has a non-empty dictionary then the student is still having unfulfilled milestones and we are returning `True`, if the dictionary is empty or the student have fulfilled the milestones then we are returning `False`.

## Notes
1. This solution won't affect the result if the learner completed the prerequisites since the `get_course_milestones_fulfillment_paths` will return an empty dictionary if the **Entrance Exam** fulfilled.
2. EdX PR: [#14758](https://github.com/edx/edx-platform/pull/14758)

Update
=====
Another issue arised as if the instructor enabled and disabled the **Prerequisite Course** in _Studio_'s Requirements section, the course became inaccessible for learner in the _dashboard_ and the _API_ views.
This happens because selecting **None** from the _dropdown_ list in _Studio_ has no functionality in the code.

### CODE REVIEW

File: common/djangoapps/util/milestones_helpers.py [hunk 1]
```diff
@@ -376,9 +376,13 @@ def any_unfulfilled_milestones(course_id, user_id):
     """ Returns a boolean if user has any unfulfilled milestones """
     if not settings.FEATURES.get('MILESTONES_APP', False):
         return False
-    return bool(
-        get_course_milestones_fulfillment_paths(course_id, {"id": user_id})
-    )
+
+    from milestones import api as milestones_api
+    fulfillment_paths = milestones_api.get_course_milestones_fulfillment_paths(course_id, {'id': user_id})
+
```
Comment (Salomari1987) [inline common/djangoapps/util/milestones_helpers.py]: Just one comment, why not use the function in this file: ```get_course_milestones_fulfillment_paths``` ?
I see in some functions the ```milestones_api.get_course_.....``` is being used, and only in one function the ```get_course_milestones_fulfillment_paths``` in the file is being used. I do not see the difference in the result on one hand, and I do not see why ```from milestones import api as milestones_api``` has to be used many times?

**Comment by Salomari1987** [comment]
@ahmedaljazzar ping

File: common/djangoapps/util/milestones_helpers.py (hunk 1 above)
Comment (ahmedaljazzar) [inline common/djangoapps/util/milestones_helpers.py]: The other methods are using the API gateway, so I prefer keeping it like this. In the same hand, the `get_course_milestones_fulfillment_paths` function is rechecking the feature flag again, a reason that tells not call it.

Let's wait for edX's review and see. But we still need to know the status of the changes if we need had to merge this before edX do.

**Comment by ahmedaljazzar** [comment]
pong.

**Comment by ahmedaljazzar** [comment]
@Salomari1987 pong.

**Comment by Salomari1987** [comment]
@ahmedaljazzar this works fine as expected, however when tried for course prerequisites, the has_access field will become false for the nonsuperuser. Please fix it also for course prerequisites.

**Comment by Salomari1987** [comment]
and rebase

**Comment by ahmedaljazzar** [comment]
@Salomari1987 Done :)

**Comment by ahmedaljazzar** [comment]
@Salomari1987, @halawa : This needs approval if you have time.

**Comment by OmarIthawi** [comment]
I'll take it!

**Comment by ahmedaljazzar** [comment]
Great!

**Pull request merged**